### PR TITLE
torch.library: Create helper function `is_functional_schema`

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #111661
* __->__ #111660

I will need this again soon.

Test Plan:
- existing tests

### CODE REVIEW

File: torch/_library/utils.py [hunk 1]
```diff
@@ -49,3 +51,43 @@ def parse_namespace(qualname: str) -> Tuple[str, str]:
...
+    """Check if the schema is functional.
+
+    An operator is functional if:
+    - it does not mutate any of its inputs
+    - it does not return a view on any of its inputs
+    - it has at least one return
+    """
+
+    # Lazy import because not all PyTorch builds have torchgen
+    from torchgen.model import FunctionSchema, SchemaKind
```
Comment (eellison) [inline torch/_library/utils.py:77]: Would it make more sense to use https://github.com/pytorch/pytorch/blob/0ea9646cdddce2339dfea998340c55323d617337/torch/csrc/jit/python/init.cpp#L1717 here ?

**Review by soulitzer — APPROVED** [review]
Re: elias's comment, I'm also curious what is different between torchgen parsers and torchscript parsers, but otherwise sounds good

**Comment by zou3519** [comment]
> Re: elias's comment, I'm also curious what is different between torchgen parsers and torchscript parsers, but otherwise sounds good

We could. torchgen parsers are the source of truth during codegen, the torchscript parsers look like a wrapper on C++ FunctionSchema to provide a similar amount of info.

torchgen isn't available in one internal config of PyTorch, so I'm not too worried about using it here. We can always update it later if it's a problem.

File: torch/_library/utils.py [hunk 2]
```diff
@@ -49,3 +51,43 @@ def parse_namespace(qualname: str) -> Tuple[str, str]:
...
+    if "." in name:
+        name, overload = name.split(".")
+    else:
+        overload = "default"
+    ns = getattr(torch.ops, namespace)
+    packet = getattr(ns, name)
+    return getattr(packet, overload)
+
+
+def is_functional_schema(schema: Any) -> bool:
```
Comment (eellison) [inline torch/_library/utils.py:67]: We should just use https://github.com/pytorch/pytorch/blob/fd209543d5dfced3146d64df3af6361fc61fb74a/torch/fx/operator_schemas.py#L114 ?

**Pull request closed without merging**